src/Data/Rational/Properties.agda

```diff
@@ -1719,6 +1720,91 @@ toℚᵘ-homo-∣-∣ (mkℚ -[1+ _ ] _ _) = *≡* refl
 ∣∣p∣∣≡∣p∣ : ∀ p → ∣ ∣ p ∣ ∣ ≡ ∣ p ∣
 ∣∣p∣∣≡∣p∣ p = 0≤p⇒∣p∣≡p (0≤∣p∣ p)
 
+------------------------------------------------------------------------
+-- Metric space
+------------------------------------------------------------------------
+
+private
+  d : ℚ → ℚ → ℚ
+  d p q = ∣ p - q ∣
+
+d-cong : Congruent _≡_ d
+d-cong = cong₂ _
+
+d-nonNegative : ∀ {p q} → 0ℚ ≤ d p q
+d-nonNegative {p} {q} = nonNegative⁻¹ _ {{∣-∣-nonNeg (p - q)}}
+
+d-definite : Definite _≡_ d
+d-definite {p} refl = cong ∣_∣ (+-inverseʳ p)
+
+d-indiscernable : Indiscernable _≡_ d
+d-indiscernable {p} {q} ∣p-q∣≡0 = begin
+  p               ≡˘⟨ +-identityʳ p ⟩
+  p - 0ℚ          ≡˘⟨ cong (_-_ p) (∣p∣≡0⇒p≡0 (p - q) ∣p-q∣≡0) ⟩
+  p - (p - q)     ≡⟨ cong (_+_ p) (neg-distrib-+ p (- q)) ⟩
+  p + (- p - - q) ≡˘⟨ +-assoc p (- p) (- - q) ⟩
+  (p - p) - - q   ≡⟨ cong₂ _+_ (+-inverseʳ p) (⁻¹-involutive q) ⟩
+  0ℚ + q          ≡⟨ +-identityˡ q ⟩
+  q               ∎
+  where
+    open ≡-Reasoning
+    open GroupProperties +-0-group
+
+d-sym : Metric.Symmetric d
+d-sym p q = begin
+  ∣ p - q ∣       ≡˘⟨ cong (λ # → ∣ # - q ∣) (⁻¹-involutive p) ⟩
+  ∣ - - p - q ∣   ≡˘⟨ cong ∣_∣ (neg-distrib-+ (- p) q) ⟩
+  ∣ - (- p + q) ∣ ≡⟨ cong (λ # → ∣ - # ∣) (+-comm (- p) q) ⟩
+  ∣ - (q - p) ∣   ≡⟨ ∣-p∣≡∣p∣ (q - p) ⟩
```

Comment:
Same here. It feels like `- (p - q) ≡ q - p` should be a lemma in `GroupProperties`?

Reply:
Yes! It's `⁻¹-anti-homo-//`